code/game/gamemodes/miniantags/demons/slaughter_demon/slaughter_demon.dm

```diff
@@ -0,0 +1,325 @@
+	maxHealth = 400
+	health = 400
+	speak = list("ire", "ego", "nahlizet", "certum", "veri", "jatkaa", "balaq", "mgar", "karazet", "geeri", "orkan", "allaq")
+	icon = 'icons/mob/mob.dmi'
+	icon_state = "daemon"
+	icon_living = "daemon"
+	obj_damage = 50
+	melee_damage_lower = 30
+	melee_damage_upper = 30
+	environment_smash = ENVIRONMENT_SMASH_RWALLS
+	armour_penetration_percentage = 100
+	move_force = MOVE_FORCE_VERY_STRONG
+	move_resist = MOVE_FORCE_VERY_STRONG
+	pull_force = MOVE_FORCE_VERY_STRONG
+	///Determines if the demon is chaneling
+	var/channeling = FALSE
+	///Used for to track health regen when phased out
+	var/last_heal_time = 0
+	///Whether or not the demon is phased into reality or not
+	var/phased = TRUE
+	///World time for when to phase out
+	var/phase_out_chanel_time = 0
+	///World time for when to phase in
+	var/phase_in_chanel_time = 0
+	///The current amplification level for damage resistance/bonus damage
+	var/stat_amplification = 0.5
+	///max % bonus to stat amp
+	var/max_amplification_stacks = 0.5
+	///Tracks the last world time since last time stat amplification was lowered
+	var/last_stat_decrease = 0
+	///World time when stat reduction will begin to tick down
+	var/decay_time = 0
+	///Bool variable whether to add speed or not
+	var/boost = 0 //Bool variable whether to add speed or not
+	var/feast_sound = 'sound/misc/demon_consume.ogg'
+	///Creatures devoured
+	var/devoured = 0
+	///Creatures with a mind absorbed total
+	var/minds_absorbed = 0
+	///List of consumed mobs
+	var/list/consumed_mobs = list()
+	var/vialspawned = FALSE
+	///Target tile for phasing in
+	var/turf/channel_target = 0
+	var/datum/spell/demon_slam/slam_holder
+	var/datum/spell/demon_charge/charge_holder
+	loot = list(/obj/effect/decal/cleanable/blood/innards, /obj/effect/decal/cleanable/blood, /obj/effect/gibspawner/generic, /obj/effect/gibspawner/generic, /obj/item/organ/internal/heart/demon/slaughter)
+	var/playstyle_string = "<B>You are the Slaughter Demon, a terrible creature from another existence. You have a single desire: to kill. \
+						       You may use the blood crawl icon when on blood pools to travel through them, appearing and disappearing from the station with a short delay. \
+						       While blood crawling you will heal back to half your health pool. Slowly. \
+						       If sink into your blood while dead or critical mobs are around you, you will pull them in with you, allowing you to feast. Healing 100HP each. \
+						       A bonus of 10 maximum hp is given per sentient humanoid consumed, every fourth corpse eaten will increase health by another 50. \
+						       Attacking sentient humanoids grants a 5% damage increase and a 5% damage reduction, stacking up to 50% extra. Maxing out at 45 damage. You will loose this buff out of combat however.<B>"
+	deathmessage = "screams in anger as it collapses into a puddle of viscera!"
+
+/mob/living/simple_animal/demon/slaughter_demon/laughter
```

Comment:
Move the laughter related stuff to after the last proc related to the slaughter demon, we usually organize it like this.